services/namex-pay/src/namex_pay/resources/worker.py

```diff
@@ -0,0 +1,359 @@
+    the Queue should be noted here:
+    - Empty or garbaled messages are knocked off the Q
+    - If the Filing is already marked paid, skip and knock off Q
+    - Once the filing is marked paid, no errors should escape to the Q
+    - If there's no matching filing, put back on Q
+    """
+    session = Session()
+    cached_session = CacheControl(session)
+
+    if len(verify_res := verify_jwt(request, cached_session)) > 0:
+        return f"{verify_res}\n", HTTPStatus.UNAUTHORIZED
+
+    structured_log(request, "INFO", f"Incoming raw msg: {request.data}")
+
+    # 1. Get cloud event
+    # ##
+    if not (ce := queue.get_simple_cloud_event(request)):
+        #
+        # Decision here is to return a 200,
+        # so the event is removed from the Queue
+        return {}, HTTPStatus.OK
+
+    structured_log(request, "INFO", f"received ce: {str(ce)}")
+    structured_log(request, "INFO", f"Incoming raw msg: {request.headers}")
+
+
+    # 2. Get payment information
+    # ##
+    if not (payment_token := get_payment_token(ce)) or payment_token.status_code != "COMPLETED":
+    # no payment info, or not a payment COMPLETED token, take off Q
+        return {}, HTTPStatus.OK
+
+    # 3. Process payment 
+    # ##
+    with current_app.app_context():
+        structured_log(request, "INFO", f"process namex payment for pay-id: {payment_token.id}")
+        process_payment(ce)
+
+    structured_log(request, "INFO", f"completed ce: {str(ce)}")
+    return {}, HTTPStatus.OK
+
+@dataclass
+class PaymentToken:
+    """Payment Token class"""
+
+    id: Optional[str] = None
+    status_code: Optional[str] = None
+    filing_identifier: Optional[str] = None
+    corp_type_code: Optional[str] = None
+
+
+def verify_jwt(request, cached_session):
+    try:
+        msg = ''
+        if current_app.config.get("DEBUG_REQUEST"):
+            structured_log(request, "INFO", f"Headers: {request.headers}")
+
+        # Get the Cloud Pub/Sub-generated JWT in the "Authorization" header.
+        bearer_token = request.headers.get("Authorization")
+        token = bearer_token.split(" ")[1]
+
+        claim = id_token.verify_oauth2_token(
+            token, Request(session=cached_session), audience=current_app.config.get("NAMEX_SUB_AUDIENCE")
+        )
+
+        if claim['email'] != current_app.config.get("NAMEX_SUB_SA_EMAIL") or not claim['email_verified']:
+            msg = f'Wrong service account email provided={current_app.config.get("NAMEX_SUB_SA_EMAIL")}'
+    except Exception as e:
+        msg = f"Invalid token: {e}\n"
+    finally:
+        return msg
+
+
+def get_payment_token(ce: SimpleCloudEvent):
+    """Return a PaymentToken if enclosed in the cloud event."""
+    if (
+        (ce.type == "payment")
+        and (data := ce.data)
+        and isinstance(data, dict)
+        and (payment_token := data.get("paymentToken", {}))
+    ):
+        converted = dict_keys_to_snake_case(payment_token)
+        pt = PaymentToken(**converted)
```

Comment:
CATTRS or some sort of other helper library probably could do this easier for you